prboom2/src/v_video.c

```diff
@@ -1440,11 +1440,29 @@ void V_DrawRawScreenSection(const char *lump_name, int source_offset, int dest_y
     }
   }
 
-  x_factor = (float)SCREENWIDTH / 320;
-  y_factor = (float)SCREENHEIGHT / 200;
-
-  if (y_factor < x_factor)
-    x_factor = y_factor;
+  switch (render_stretch_hud) {
+    case patch_stretch_not_adjusted:
+      x_factor = (float)SCREENWIDTH / 320;
+      y_factor = (float)SCREENHEIGHT / 200;
+      if (y_factor < x_factor)
+        x_factor = y_factor;
+      break;
+    case patch_stretch_doom_format:
+      x_factor = (float)WIDE_SCREENWIDTH / 320;
+      y_factor = (float)WIDE_SCREENHEIGHT / 200;
+      if (y_factor < x_factor)
+        x_factor = y_factor;
+      break;
+    case patch_stretch_fit_to_width:
+      x_factor = (float)SCREENWIDTH / 320;
+      y_factor = (float)SCREENHEIGHT / 200;
+      break;
+    default:
+      // This should never happen
+      x_factor = 0;
+      y_factor = 0;
+      break;
+  }
```

Comment:
No need for the `default` path

Reply:
Only reason I added a default path was to fix GCC warnings.

Since I've started working on DSDA Doom code, it's been special in that GCC had no warnings in the continuous integration, and so whenever I work with the code, I try and keep it that way.

Reply:
All other instances of this switch statement are not using `default`. Thats why the comment.
And idk why the CI would not give you any warning, there are tons of warnings on my machine

Reply:
What I'm particularly referencing is the GCC Linux and MSYS2 build log from the GitHub continuous integration.

When I started working on DSDA code these showed absolutely no warnings during the build process and so I tend to try and keep it that way.

I agree I wasn't crazy about using `default` there, I was just finding a way to squash a possible undefined warning relating to `x_factor` and `y_factor`.